web/modules/custom/joinup_core/src/Plugin/Action/ChangeGroupAction.php

```diff
@@ -0,0 +1,191 @@
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\EntityReferenceSelection\SelectionPluginManagerInterface $selection_plugin_manager
+   *   The selection plugin manager service.
+   * @param \Drupal\Core\Render\Renderer $renderer
+   *   The renderer service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, SelectionPluginManagerInterface $selection_plugin_manager, Renderer $renderer) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->selectionPluginManager = $selection_plugin_manager;
+    $this->renderer = $renderer;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('plugin.manager.entity_reference_selection'),
+      $container->get('renderer')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function access($node, AccountInterface $account = NULL, $return_as_object = FALSE) {
+    // The access is limited at the view level.
+    return $return_as_object ? AccessResult::allowed() : TRUE;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function executeMultiple(array $nodes) {
+    $message = array_fill_keys(['error', 'warning', 'status'], []);
+    $group_id = $this->getConfiguration()['group'];
+    if (!$group = Rdf::load($group_id)) {
+      throw new \RuntimeException("Cannot load RDF entity with ID $group_id");
+    }
+
+    /** @var \Drupal\node\NodeInterface[] $nodes */
+    foreach ($nodes as $node) {
+      $is_custom_page = $node->bundle() === 'custom_page';
+      $args = [
+        '%title' => $node->label(),
+        '%group' => $group->label(),
+        '@type' => $node->type->entity->label(),
+      ];
+      // Custom pages cannot be group content in solutions.
+      if ($group->bundle() === 'solution' && $is_custom_page) {
+        $message['warning'][] = $this->t("Group of %title cannot be changed because a custom page cannot be be attached to a solution.", $args);
+        continue;
+      }
+      try {
+        if ($is_custom_page) {
+          // Custom pages should create disabled menu items.
+          // @see \Drupal\custom_page\CustomPageOgMenuLinksUpdater::addLink()
+          $node->exclude_from_menu = TRUE;
+        }
+        // Prevent notification dispatching.
+        // @see joinup_notification_dispatch_notification()
+        $node->skip_notification = TRUE;
+        $node->set('og_audience', $group_id)->save();
+
+        $status = ['#markup' => $this->t("The group of @type '%title' has been changed to '%group'.", $args)];
+        if ($is_custom_page) {
+          $status[] = [
+            '#prefix' => ' ',
+            '#markup' => $this->t('The custom page menu link is disabled in the new group and it should be manually enabled.'),
```

Comment:
Is this a functional requirement to make the custom pages disabled by default? I don't see this mentioned in the original ticket.